pkg/cloudprovider/openstack/openstack.go

```diff
@@ -625,28 +626,41 @@ func (lb *LoadBalancer) UpdateTCPLoadBalancer(name, region string, hosts []strin
 func (lb *LoadBalancer) DeleteTCPLoadBalancer(name, region string) error {
 	glog.V(4).Infof("DeleteTCPLoadBalancer(%v, %v)", name, region)
 
-	vip, err := getVipByName(lb.network, name)
-	if err != nil {
-		return err
+	vip, vipErr := getVipByName(lb.network, name)
+	vipExists := true
+	if vipErr != nil {
+		if vipErr != ErrNotFound {
+			return vipErr
+		}
+		vipExists = false
 	}
 
-	pool, err := pools.Get(lb.network, vip.PoolID).Extract()
-	if err != nil {
-		return err
+	pool, poolErr := pools.Get(lb.network, vip.PoolID).Extract()
+	poolExists := true
+	if poolErr != nil {
+		detailedErr, ok := poolErr.(*gophercloud.UnexpectedResponseCodeError)
+		if !ok || detailedErr.Actual != http.StatusNotFound {
+			return poolErr
+		}
+		poolExists = false
 	}
 
 	// Have to delete VIP before pool can be deleted
-	err = vips.Delete(lb.network, vip.ID).ExtractErr()
-	if err != nil {
-		return err
+	if vipExists {
+		err := vips.Delete(lb.network, vip.ID).ExtractErr()
+		if err != nil {
+			return err
+		}
 	}
 
 	// Ignore errors for everything following here
 
-	for _, monId := range pool.MonitorIDs {
-		pools.DisassociateMonitor(lb.network, pool.ID, monId)
+	if poolExists {
+		for _, monId := range pool.MonitorIDs {
+			pools.DisassociateMonitor(lb.network, pool.ID, monId)
+		}
+		pools.Delete(lb.network, pool.ID)
```

Comment:
> Since a couple weeks ago, load balancers have been named using their service's UID, so there will never be name conflicts that enable reusing old pools.

(thinking)

What about:
- Create() creates pool successfully, fails to create vip, fails to delete new pool during unwind
- Create() now gets called again.

Do we now have a duplicate pool?  Or will Create() call Delete repeatedly in some way before reattempting the Create?